frontend/src/host_orchestrator/orchestrator/artifacts/artifacts.go

```diff
@@ -101,12 +90,12 @@ func (h *Manager) GetKernelBundle(buildID, target string, fetcher Fetcher) (stri
 		}
 		return outDir, nil
 	}
-	return h.syncDownload(buildID+target+"kernel", f)
+	return h.registry.Get(outDir).Run(f)
 }
 
 func (h *Manager) GetBootloaderBundle(buildID, target string, fetcher Fetcher) (string, error) {
+	outDir := fmt.Sprintf("%s/%s_%s__bootloader", h.rootDir, buildID, target)
 	f := func() (string, error) {
-		outDir := fmt.Sprintf("%s/%s_%s__bootloader", h.rootDir, buildID, target)
 		if err := createDir(outDir); err != nil {
```

Comment:
nit: I'd prefer `outDir` to be a parameter of the functor instead of a capture.

Reply:
done